plugins/backup/dummy/src/main/java/org/apache/cloudstack/backup/DummyBackupProvider.java

```diff
@@ -0,0 +1,136 @@
+
+import org.apache.cloudstack.backup.dao.BackupDao;
+import org.apache.log4j.Logger;
+
+import com.cloud.utils.Pair;
+import com.cloud.utils.component.AdapterBase;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.vm.VirtualMachine;
+
+public class DummyBackupProvider extends AdapterBase implements BackupProvider {
+
+    private static final Logger s_logger = Logger.getLogger(DummyBackupProvider.class);
+
+    @Inject
+    private BackupDao backupDao;
+
+    @Override
+    public String getName() {
+        return "dummy";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Dummy Backup Plugin";
+    }
+
+    @Override
+    public List<BackupOffering> listBackupOfferings(Long zoneId) {
+        s_logger.debug("Listing backup policies on Dummy B&R Plugin");
+        BackupOffering policy1 = new BackupOfferingVO(1, "gold-policy", "dummy", "Golden Policy", "Gold description", true);
+        BackupOffering policy2 = new BackupOfferingVO(1, "silver-policy", "dummy", "Silver Policy", "Silver description", true);
+        return Arrays.asList(policy1, policy2);
+    }
+
+    @Override
+    public boolean isBackupOffering(Long zoneId, String uuid) {
+        s_logger.debug("Checking if backup offering exists on the Dummy Backup Provider");
+        return true;
+    }
+
+    @Override
+    public boolean assignVMToBackupOffering(VirtualMachine vm, BackupOffering backupOffering) {
+        s_logger.debug("Creating VM backup for VM " + vm.getInstanceName() + " from backup offering " + backupOffering.getName());
+        return true;
+    }
+
+    @Override
+    public boolean restoreVMFromBackup(VirtualMachine vm, Backup backup) {
+        s_logger.debug("Restoring vm " + vm.getUuid() + "from backup " + backup.getUuid() + " on the Dummy Backup Provider");
+        return true;
+    }
+
+    @Override
+    public Pair<Boolean, String> restoreBackedUpVolume(Backup backup, String volumeUuid, String hostIp, String dataStoreUuid) {
+        s_logger.debug("Restoring volume " + volumeUuid + "from backup " + backup.getUuid() + " on the Dummy Backup Provider");
+        throw new CloudRuntimeException("Dummy plugin does not support this feature");
+    }
+
+    @Override
+    public Map<VirtualMachine, Backup.Metric> getBackupMetrics(Long zoneId, List<VirtualMachine> vms) {
+        final Map<VirtualMachine, Backup.Metric> metrics = new HashMap<>();
+        final Backup.Metric metric = new Backup.Metric(1000L, 100L);
+        for (VirtualMachine vm : vms) {
+            metrics.put(vm, metric);
+        }
+        return metrics;
+    }
+
+    @Override
+    public boolean removeVMFromBackupOffering(VirtualMachine vm) {
+        s_logger.debug("Removing VM ID " + vm.getUuid() + " from backup offering by the Dummy Backup Provider");
+        final List<Backup> backups = backupDao.listByVmId(null, vm.getId());
+        for (final Backup backup : backups) {
+            backupDao.remove(backup.getId());
```

Comment:
If assignVMToBackupOffering does nothing then why we try to remove the VM here? Can we just return true as well?

Reply:
the idea the remove VM from offering should also remove all the backups, which is why we're deleting/removing all the backups of the VM when it's removed from the offering.